src/main/java/org/wso2/carbon/connector/amazonsqs/utils/Utils.java

```diff
@@ -0,0 +1,262 @@
+import software.amazon.awssdk.services.sqs.model.MessageAttributeValue;
+import software.amazon.awssdk.services.sqs.model.MessageSystemAttributeName;
+import software.amazon.awssdk.services.sqs.model.MessageSystemAttributeNameForSends;
+import software.amazon.awssdk.services.sqs.model.MessageSystemAttributeValue;
+import software.amazon.awssdk.services.sqs.model.SendMessageRequest;
+import software.amazon.awssdk.services.sqs.model.SqsResponseMetadata;
+
+import javax.xml.stream.XMLStreamException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This class contain required util methods for to Amazon sqs connector.
+ */
+public class Utils {
+
+    private static final Log log = LogFactory.getLog(Utils.class);
+
+    public static OMElement createOMElement(String elementName, Object value) {
+        OMElement resultElement = null;
+        try {
+            if (value != null) {
+                resultElement = AXIOMUtil.
+                        stringToOM("<" + elementName + ">" + value
+                                + "</" + elementName + ">");
+            } else {
+                resultElement = AXIOMUtil.
+                        stringToOM("<" + elementName
+                                + "></" + elementName + ">");
+            }
+        } catch (XMLStreamException | OMException e) {
+            log.error("Error while generating OMElement from element name" + elementName, e);
+        }
+        return resultElement;
+    }
+
+    public static String createUrl(MessageContext messageContext, SqsConnection sqsConnection)
+            throws MalformedURLException, SqsInvalidConfigurationException {
+        try {
+            String queueName = (String) ConnectorUtils.lookupTemplateParamater(messageContext, Constants.QUEUE_NAME);
+            String queueId = (String) ConnectorUtils.lookupTemplateParamater(messageContext, Constants.QUEUE_ID);
+            String queueUrl = (String) ConnectorUtils.lookupTemplateParamater(messageContext, Constants.QUEUE_URL);
+            if (StringUtils.isNotBlank(queueUrl)) {
+                return queueUrl;
+            }
+            if (!StringUtils.isNotBlank(queueName) && (!StringUtils.isNotBlank(queueId) ||
+                    !StringUtils.isNotBlank(queueUrl))) {
+                throw new SqsInvalidConfigurationException("Missing queue info: Should provide value of queue url or " +
+                        "queue name and queue id.");
+            }
+            return new URL(Constants.HTTPS, "sqs".concat(sqsConnection.getConnectionConfig().getRegion()).
+                    concat(".amazonaws.com"), Constants.SLASH.concat(queueId).
+                    concat(Constants.SLASH).concat(queueName)).toString();
+        } catch (MalformedURLException e) {
+            throw new MalformedURLException(e.getMessage());
```

Comment:
Shall we throw a SqsInvalidConfigurationException from here as `MalformedURLException` is thrown if it's an issue with the given configs